src/actors/process_list/process_list.py

```diff
@@ -0,0 +1,76 @@
+import os
+import sys
+import json
+
+
+def readlink(path, filename):
+    return os.readlink(os.path.join(path, filename))
+
+
+def get_environ(pid):
+    try:
+        with open(os.path.join(pid, 'environ'), 'rb') as f:
+            output = {}
+            environ = f.read().split("\x00")
+            for env in environ:
+                if env:
+                    name, val = env.split('=', 1)
+                    output[name] = val
+            return output
+    except EnvironmentError as exc:
+        sys.stderr.write(str(exc) + "\n")
+        return None
+
+
+def get_cmdline(pid):
+    try:
+        with open(os.path.join(pid, 'cmdline'), 'rb') as f:
+            return f.read().replace("\x00", " ").strip()
+    except EnvironmentError as exc:
+        sys.stderr.write(str(exc) + "\n")
+        return None
+
+
+def extract_filter_param(params):
+    if params:
+        try:
+            return str(params['process_list_filter'][0]['value'])
+        except LookupError:
+            sys.stderr.write("Invalid filter param structure, continue without filtering...\n")
+    return ''
+
+
+if __name__ == '__main__':
+    params = {} if sys.stdin.isatty() else json.load(sys.stdin)
+    filter_param = extract_filter_param(params)
+
+    actor_output = {'process_list': [
+        {'processes': []}
+    ]}
+
+    root = '/proc'
+    cpid = str(os.getpid())
+
+    for directory in os.listdir(root):
+
+        if not directory.isdigit() or directory == cpid:
+            continue
+
+        pid = os.path.join(root, directory)
+        cmdline = get_cmdline(pid)
+        environ = get_environ(pid)
+
+        if cmdline is None or environ is None:
+            continue
+
+        if filter_param not in cmdline:
+            continue
```

Comment:
Would it make sense to split `cmdline` and match the whole last element? 

I think we might have a problem in this situation: let's say I have a Tornado app running in my machine and I'm tailing the logs with `tailf -f /my/app/dir/tornado.log` in a separate screen/tmux section. If `filter_param == "tornado"`, the execution flow will not be interrupted here and the actor will return the wrong information.

Reply:
i fixed that by regex, in that case i can't split cmdline